packages/wallet/backend/src/rafiki/service.ts

```diff
@@ -391,55 +395,126 @@ export class RafikiService implements IRafikiService {
+        }`
+      )
+      this.deps.logger.info(
+        `Send amount value: ${receivedQuote.sendAmount.value}`
+      )
+      this.deps.logger.info(
+        `Receive amount: ${receivedQuote.receiveAmount.value} from rafiki which includes cross currency fees already.`
+      )
+    }
+
+    const actualFee = feeStructure[receivedQuote.sendAmount.assetCode]
+
     if (receivedQuote.paymentType == PaymentType.FixedDelivery) {
       if (
-        receivedQuote.sendAmount.assetCode !== feeStructure.asset ||
-        receivedQuote.sendAmount.assetScale !== feeStructure.scale
+        feeStructure[receivedQuote.sendAmount.assetCode] &&
+        receivedQuote.sendAmount.assetScale !==
+          feeStructure[receivedQuote.sendAmount.assetCode].scale
       ) {
         throw new BadRequest('Invalid quote sendAmount asset')
       }
       const sendAmountValue = BigInt(receivedQuote.sendAmount.value)
+
       const fees =
         // TODO: bigint/float multiplication
-        BigInt(Math.floor(Number(sendAmountValue) * feeStructure.percentage)) +
-        BigInt(feeStructure.fixed)
+        BigInt(Math.floor(Number(sendAmountValue) * actualFee.percentage)) +
+        BigInt(actualFee.fixed)
+
+      this.deps.logger.info(
+        `wallet fees: (sendAmount (${Math.floor(
+          Number(sendAmountValue)
+        )}) * wallet percentage (${actualFee.percentage})) + fixed ${
+          actualFee.fixed
+        } = ${fees}`
+      )
 
       receivedQuote.sendAmount.value = sendAmountValue + fees
+
+      this.deps.logger.info(
+        `Will finally send: ${receivedQuote.sendAmount.value}`
+      )
     } else if (receivedQuote.paymentType === PaymentType.FixedSend) {
-      if (receivedQuote.receiveAmount.assetCode !== feeStructure.asset) {
+      if (
+        !Object.keys(feeStructure).includes(
+          receivedQuote.receiveAmount.assetCode
+        )
+      ) {
         throw new BadRequest('Invalid quote receiveAmount asset')
       }
+
       const receiveAmountValue = BigInt(receivedQuote.receiveAmount.value)
+
       const fees =
-        BigInt(
-          Math.floor(Number(receiveAmountValue) * feeStructure.percentage)
-        ) + BigInt(feeStructure.fixed)
+        BigInt(Math.floor(Number(receiveAmountValue) * actualFee.percentage)) +
+        BigInt(actualFee.fixed)
+
+      this.deps.logger.info(
+        `Wallet fee: ${Math.floor(Number(receiveAmountValue))}  * ${
+          actualFee.percentage
+        }  + fixed: ${BigInt(actualFee.fixed)}  = ${fees}`
+      )
 
       if (receiveAmountValue <= fees) {
         throw new BadRequest('Fees exceed quote receiveAmount')
       }
 
       receivedQuote.receiveAmount.value = receiveAmountValue - fees
+
+      this.deps.logger.info(
+        `Sum of fees (conversion fee from rafiki + wallet fee): ${
+          receivedQuote.sendAmount.value - receivedQuote.receiveAmount.value
+        } + ${fees} ${receiveAmountValue - fees}`
+      )
+
+      this.deps.logger.info(
```

Comment:
Should we only log these in the development environment?

Reply:
yeah I think so. added dev check